operations/network/dns/PowerDNS_Troubleshooting.md

```diff
@@ -0,0 +1,141 @@
+# PowerDNS troubleshooting
+
+## List DNS zone contents
+
+The PowerDNS zone database is populated with data from two sources
+
+* The cray-powerdns-manager service creates the zones and DNS records based on data sourced from the System Layout Service (SLS)
+* The external DNS records are populated by the cray-externaldns-external-dns service using data sourced from Kubernetes annotations and virtual service definitions.
+
+The `cray-powerdns-visualizer` command can be used to view the zone structure that cray-powerdns-manager will create.
+
+```
+ncn-m001:~ # kubectl -n services exec deployment/cray-powerdns-manager -c cray-powerdns-manager -- cray-powerdns-visualizer
+.
+├── 252.10.in-addr.arpa.
+│   ├── [PTR]  5.252.10.in-addr.arpa.
+│   │   └── sw-leaf-002.nmn.wasp.dev.cray.com.
+│   ├── [PTR]  4.252.10.in-addr.arpa.
+│   │   └── sw-leaf-001.nmn.wasp.dev.cray.com.
+│   ├── [PTR]  3.252.10.in-addr.arpa.
+│   │   └── sw-spine-002.nmn.wasp.dev.cray.com.
+│   ├── [PTR]  6.2.252.10.in-addr.arpa.
+│   │   └── pbs_comm_service.nmn.wasp.dev.cray.com.
+│   ├── [PTR]  5.2.252.10.in-addr.arpa.
+│   │   └── pbs_service.nmn.wasp.dev.cray.com.
+│   ├── [PTR]  4.2.252.10.in-addr.arpa.
+│   │   └── slurmdbd_service.nmn.wasp.dev.cray.com.
+│   ├── [PTR]  3.2.252.10.in-addr.arpa.
+│   │   └── slurmctld_service.nmn.wasp.dev.cray.com.
+│   ├── [PTR]  2.2.252.10.in-addr.arpa.
+│   │   └── uai_macvlan_bridge.nmn.wasp.dev.cray.com.
+...
+```
+
+For more information on External DNS and troubleshooting steps see the [External DNS documentation](../external_dns/External_DNS.md)
+
+## PowerDNS logging
+
+When troubleshooting DNS problems it may prove helpful to increase the level of logging from the default value of 3 (error).
+
+1. Edit the cray-dns-powerdns ConfigMap
+
+   ```
+   ncn-m001:~ # kubectl -n services edit cm cray-dns-powerdns
+   ```
+
+1. Set the `loglevel` parameter in `pdns.conf` to the desired setting
+
+   ```
+   pdns.conf: |
+      config-dir=/etc/pdns
+      include-dir=/etc/pdns/conf.d
+      guardian=yes
+      loglevel=3
+      setgid=pdns
+      setuid=pdns
+      socket-dir=/var/run
+      version-string=anonymous
+   ```
+
+1. Restart the PowerDNS service
+
+   ```
+   ncn-m001:~ # kubectl -n services rollout restart deployment cray-dns-powerdns
+   deployment.apps/cray-dns-powerdns restarted
+   ```
+
+Please see the [PowerDNS documentation](https://doc.powerdns.com/authoritative/settings.html#loglevel) for more information.
```

Comment:
Change the beginning of the sentence to "Refer to the external..."

Reply:
Fixed